v2/moviebuff.go

```diff
@@ -0,0 +1,336 @@
+// moviebuff-sdk-go is Go SDK for Moviebuff.
+//
+// The Moviebuff is a service that offers information about movies, people and entities.
+//
+// All resources are identified by a UUID, which Moviebuff.com uniquely and randomly generates.
+// Since it may be difficult to get the Moviebuff UUID of any resource without prior knowledge,
+// the API also allows substitution of the UUID with the URL identifier of the resource.
+// For instance, if when attempting to get information for 12 Years a Slave (moviebuff.com/12-years-a-slave),
+// either its UUID or the identifier 12-years-a-slave may be used.
+package moviebuff
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+)
+
+// Type of resources supported by moviebuff
+type ResourceType string
+
+const (
+	RESOURCE_TYPE_PEOPLE   ResourceType = "people"
+	RESOURCE_TYPE_MOVIES   ResourceType = "movies"
+	RESOURCE_TYPE_ENTITIES ResourceType = "entities"
+)
+
+var (
+	ErrInvalidToken         = errors.New("access denied")
+	ErrResponseNotReceived  = errors.New("could not receive valid response")
+	ErrResourceDoesNotExist = errors.New("resource does not exist")
+)
+
+// Moviebuff allows to access to information in moviebuff using resource ids.
+type Moviebuff interface {
+	GetMovie(ctx context.Context, id string) (*Movie, error)
+	GetPerson(ctx context.Context, id string) (*Person, error)
+	GetEntity(ctx context.Context, id string) (*Entity, error)
+	GetResources(ctx context.Context, resourceType ResourceType, limit, page int) (*Resources, error)
+	GetCertifications(ctx context.Context, country string) ([]Certification, error)
+	GetHolidayCalendar(ctx context.Context, countryID string) (*Calendar, error)
+}
+
+type Config struct {
+	HostURL     string
+	StaticToken string
+	Client      *http.Client
+}
+
+// Before accessing any API it need to be initialized.
+// The Moviebuff is a service that offers information about movies, people, entities.
+type moviebuff struct {
+	Config
+}
+
+// New returns a Moviebuff interface.
+func New(config Config) Moviebuff {
+	if config.Client == nil {
+		config.Client = http.DefaultClient
+	}
+	return &moviebuff{
+		Config: config,
+	}
+}
+
+// GetMovie fetch a movie and its basic details for given resource uuid.
+
+// Instead of the UUID, this can also be the URL of the movie as seen on moviebuff.com, like 12-years-a-slave
+// Details include release dates, certifications, cast, crew, trailers, posters, purchase links etc.
+// Here movies may include feature films, documentaries, short films etc.
+func (m *moviebuff) GetMovie(ctx context.Context, id string) (*Movie, error) {
+	r, err := prepareRequest(ctx, m.HostURL, m.StaticToken, "/resources/movies/"+id)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := m.Client.Do(r)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
```

Comment:
There's a lot of repeated code from v1. Since we're importing v1 anyway, can we move common code? This will also make adding new functions easy